packages/react-native-web/src/vendor/react-native/Utilities/useCallbackRef.js

```diff
@@ -0,0 +1,16 @@
+/*
+ *  Helper to keep the same ref of a function accross renders 
+ * 
+ */
+import { useLayoutEffect, useEffect, useRef } from 'react';
+
+export default function useCallbackRef(callback) {
+  const callbackRef = useRef(callback);
+  const useIsomorphicLayoutEffect = typeof window !== 'undefined' ? useLayoutEffect : useEffect;
+
+  useIsomorphicLayoutEffect(() => {
```

Comment:
Is there any point delaying the ref update through an effect?

Reply:
initial ref is set with `const callbackRef = useRef(callback);`

it then updates in the effect

Reply:
I understand that. I mean, you can simply update the ref without any effects. Simpler code and it guarantees that the most recent value of `callback` is assigned to the ref and will be called by Image - you don't know when exactly the effect will be executed and Image may stop loading between the render and the effect.

Reply:
That's true. I'm wondering why `swr` does it [this](https://github.com/vercel/swr/blob/main/src/index/use-swr.ts#L575) way though? (They also don't have the config in the dependency array). There's also the `experimental_useEffectEvent` from react, so I guess there are edge cases with both solutions